javascript-source/core/chatModerator.js

```diff
@@ -494,10 +495,24 @@
      * @param {string} message
      */
     function checkWhiteList(message) {
+        function checkLink(link, whiteListItem) {
+            var baseLink = link.match(/[^.]*[^/]*/)[0];
+            var itemRe = RegExp(whiteListItem.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&'), 'g');
```

Comment:
I'm not sure if in NodeJS not using `new` on a class works, but in Rhino it will cause an error for sure.